course_discovery/apps/course_metadata/models.py

```
@@ -1503,6 +1507,21 @@ class Meta(object):
         )
         ordering = ['created']
 
+    def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
+        super(Seat, self).save(*args, **kwargs)
+        if self.upgrade_deadline and self.upgrade_deadline != self.__original_upgrade_deadline:
+            # Avoid circular imports
+            from course_discovery.apps.publisher.models import CourseRun as PublisherCourseRun
+            try:
+                run = PublisherCourseRun.objects.get(lms_course_id=self.course_run.key)
+                seat = run.seats.get(type=self.type)
+                seat.upgrade_deadline = self.upgrade_deadline
+                seat.save()
+            except ObjectDoesNotExist as exc:
+                logger.warning(
```

`logger.exception` would cause the stacktrace to be printed, which is usually helpful for figuring out where in the code a message is coming from / why it's happening.

But more importantly, why is it an error if the publisher model doesn't exist? The seat could be from before publisher existed. The seat could be from the new publisher frontend, which doesn't keep its data in the old publisher models. I think you should just silently ignore it if no publisher model exists for this seat/run.